packages/core/lib/Client.ts

```diff
@@ -0,0 +1,136 @@
+import * as Adapter from './Adapter';
+import { Event, IBreadcrumb, IUser, LogLevel } from './Interfaces';
+import * as Context from './Interfaces/Context';
+import { DSN } from './Interfaces/DSN';
+import { IOptions } from './Options';
+import { SentryError } from './Sentry';
+
+// TODO: Add breadcrumbs
+// TODO: Add context handling tags, extra, user
+export class Client {
+  public readonly dsn: DSN;
+  private _adapter: Adapter.IAdapter;
+  private _context: Context.IContext;
+
+  constructor(
+    dsn: string,
+    public options: IOptions = {
+      logLevel: LogLevel.Error,
+      maxBreadcrumbs: 100,
+    }
+  ) {
+    this.dsn = new DSN(dsn);
+    this._context = Context.getDefaultContext();
+    return this;
+  }
+
+  public getContext() {
+    // TODO: check for cyclic objects
+    return JSON.parse(JSON.stringify(this._context));
+  }
+
+  private get adapter() {
+    if (!this._adapter) {
+      throw new SentryError('No adapter in use, please call .use(<Adapter>)');
+    }
+    return this._adapter;
```

Comment:
I'm questioning this getter here, is there a really strong reason to have it instead of a regular function `adapter()`?

Reply:
So the fancy thing about this is that I can write `this.adapter` everywhere in my code and have a check if an adapter is set. But yeah, it is not necessary ...

Reply:
Sure, but you could do `this.adapter().blabla` anywhere in the code as well and it's a bit more expressive. Otherwise someone making a quick change to this file might miss that part. 
Anyway, I'll leave it up to you.